schematic/schematics/redshift_schematic.py

```
@@ -0,0 +1,553 @@
+        """Determine if value can be inserted into column of
+           the group of types described by the class.
+
+        Args:
+          value: The value to check.
+        """
+        try:
+            cast_value = float(value)
+        except ValueError:
+            return False
+        return (cast_value >= self.min_value and
+                cast_value <= self.max_value and
+                cast_value // 1 == cast_value)
+
+
+class RedshiftBigIntType(RedshiftAbstractIntType):
+    """An bigint type in Redshift"""
+    name = "RedshiftBigIntType"
+    next_less_restrictive = RedshiftFloatType
+    min_value = -9223372036854775808
+    max_value = 9223372036854775807
+
+    def __init__(self):
+        super(RedshiftBigIntType, self).__init__()
+
+    def to_sql(self):
+        return "BIGINT".format(self.len)
+
+
+class RedshiftIntType(RedshiftAbstractIntType):
+    """An int type in Redshift"""
+    name = "RedshiftIntType"
+    next_less_restrictive = RedshiftBigIntType
+    parameterized = False
+    min_value = -2147483648
+    max_value = 2147483647
+
+    def __init__(self):
+        super(RedshiftIntType, self).__init__()
+
+    def to_sql(self):
+        return "INT".format(self.len)
+
+
+class RedshiftSmallIntType(RedshiftAbstractIntType):
+    """A smallint type in Redshift"""
+    name = "RedshiftSmallIntType"
+    next_less_restrictive = RedshiftIntType
+    parameterized = False
+    min_value = -32768
+    max_value = 32767
+
+    def __init__(self):
+        super(RedshiftSmallIntType, self).__init__()
+
+    def to_sql(self):
+        return "SMALLINT".format(self.len)
+
+
+class RedshiftTableDefinition(schematic.TableDefinition):
+    """Redshift-specific implementation of TableDefinition"""
+
+    def __init__(self, schema, name, columns):
+        self.sortkeys = []
+        sk_dict = {}
+        self.distkey = None
+        for col in columns:
+            if col.distkey:
+                self.distkey = col
```

probably want to raise an error if we encounter more than one distkey

---

Good call, updated.